ldap/servers/slapd/back-ldbm/ldbm_modrdn.c

```diff
@@ -23,6 +23,8 @@ static IDList *moddn_get_children(back_txn *ptxn, Slapi_PBlock *pb, backend *be,
 static int moddn_rename_children(back_txn *ptxn, Slapi_PBlock *pb, backend *be, IDList *children, Slapi_DN *dn_parentdn, Slapi_DN *dn_newsuperiordn, struct backentry *child_entries[]);
 static int modrdn_rename_entry_update_indexes(back_txn *ptxn, Slapi_PBlock *pb, struct ldbminfo *li, struct backentry *e, struct backentry **ec, Slapi_Mods *smods1, Slapi_Mods *smods2, Slapi_Mods *smods3, Slapi_Mods *smods4);
 static void mods_remove_nsuniqueid(Slapi_Mods *smods);
+static int32_t dsentrydn_modrdn_update(backend *be, const char *olddn, const char *newdn, struct backentry *e, struct backentry **ec, ID id, back_txn *txn);
+static int dsentrydn_moddn_rename(back_txn *ptxn, backend *be, ID id, IDList *children, const Slapi_DN *dn_parentdn, const Slapi_DN *dn_newsuperiordn, struct backentry *child_entries[]);
```

Comment:
In a c file, functions can only be called if the declaration is "before" the first usage. Since dsentrydn_moddn_rename_child's full function impl is *before* it's called at all, there is no need for the declaration here. 

Remember a c file is "one pass" from top to bottom, and if a "name" isn't defined it's an error. So the following would fail.

```

void foo() {
    bar()
}

void bar() {
}
```

Because as void foo is read, bar() isn't defined yet. These both below would work.

```
void bar(); 

void foo() {
    bar()
}

void bar() {
}
```

```
void bar() {
}

void foo() {
    bar()
}
```

Hope that helps :)